### PR TITLE
add Add Strings

### PR DESCRIPTION
**Pull request by tkkawa**
Please check my code.

### CODE REVIEW

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
See this:

Follow up: Could you solve it without using any built-in BigInteger library or converting the inputs to integer directly?

File: 415_add-strings.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        res = []
+        a = 0
+        for i in range(min(len(lis1), len(lis2))):
+            res.append(num.index(lis1[i]) + num.index(lis2[i]))
+            if i == len(lis1)-1:
+                lis = lis2
+            if i == len(lis2)-1:
+                lis = lis1
+        if len(lis) != len(res):
+            for i in range(len(res), len(res)+max(len(lis1), len(lis2))-min(len(lis1), len(lis2))):
```
Comment (ryuji0123) [inline 415_add-strings.py:19]: too difficult to understand

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
Basically, you should avoid complex codes.
See solution in leetcode

**Comment by tkkawa** [comment]
It's solution in leetcode.

File: 415_add-strings.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# n = max(num2dd1.length, num2.length)
+# time : O(n)
+# space : O(1)
+class Solution:
+    def addStrings(self, num1: str, num2: str) -> str:
+        total1 = ord(num1[0])-48
```
Comment (ryuji0123) [inline 415_add-strings.py:6]: avoid using magic number (in this case, 48's meaning is very unclear) again.

File: 415_add-strings.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        for i in num1[1:]:
+            num = ord(i)-48
+            total1 = total1*10+num
+        total2 = ord(num2[0])-48
+        for i in num2[1:]:
+            num = ord(i)-48
+            total2 = total2*10+num
+        total = total1+total2
```
Comment (ryuji0123) [inline 415_add-strings.py:14]: See Google style before creating PR
```
total = total1 + total2
```

File: 415_add-strings.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# n = max(num2dd1.length, num2.length)
+# time : O(n)
+# space : O(1)
+class Solution:
+    def addStrings(self, num1: str, num2: str) -> str:
+        total1 = ord(num1[0])-48
+        for i in num1[1:]:
+            num = ord(i)-48
+            total1 = total1*10+num
```
Comment (ryuji0123) [inline 415_add-strings.py:9]: See PEP8: Whitespace in Expressions and Statements

total1 = total1*10 + num

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
Check your coding style before creating PR

File: 415_add-strings.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    def addStrings(self, num1: str, num2: str) -> str:
+        zero_code = ord("0")
+        total1 = ord(num1[0]) - zero_code
+        for i in num1[1:]:
+            num = ord(i) - zero_code
+            total1 = total1 * 10 + num
+        total2 = ord(num2[0]) - zero_code
+        for i in num2[1:]:
+            num = ord(i) - zero_code
+            total2 = total2 * 10+num
```
Comment (ryuji0123) [inline 415_add-strings.py:17]: See https://www.python.org/dev/peps/pep-0008/#other-recommendations
total2 = total2*10 + num

File: 415_add-strings.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class Solution:
+
+    def addStrings(self, num1: str, num2: str) -> str:
+        zero_code = ord("0")
+        total1 = ord(num1[0]) - zero_code
+        for i in num1[1:]:
+            num = ord(i) - zero_code
+            total1 = total1 * 10 + num
```
Comment (ryuji0123) [inline 415_add-strings.py:13]: See https://www.python.org/dev/peps/pep-0008/#other-recommendations
 total1 = total1*10 + num

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Review by ryuji0123 — APPROVED** [review]
LGTM

**Pull request merged**